watchtower-plugin/src/retrier.rs

```diff
@@ -1143,16 +1173,21 @@ mod tests {
             .is_running());
 
         // Wait for the remaining time and re-check
-        tokio::time::sleep(Duration::from_secs_f64(MAX_RUN_TIME + HALF_API_DELAY)).await;
-        let state = wt_client.lock().unwrap();
-        assert!(!state.retriers.contains_key(&tower_id));
+        wait_until!(wt_client
+            .lock()
+            .unwrap()
+            .get_retrier_status(&tower_id)
+            .is_none());
 
-        let tower = state.towers.get(&tower_id).unwrap();
-        assert!(tower.status.is_reachable());
-        assert!(tower.pending_appointments.is_empty());
+        {
+            let state = wt_client.lock().unwrap();
+            assert!(!state.retriers.contains_key(&tower_id));
```

Comment:
Same here, we asserting on tower_id not in retriers but we already waited on it.